### ISSUE TITLE
Entropy Loss Calculation Question in ppo.py

### ISSUE THREAD

**Issue by Ian-Sy-Zhang** [DLR-RM/stable-baselines3#1856]
### Body:

There appears to be a problem in the entropy loss calculation in the ppo.py of Stable-baseline3. The current implementation is using the following line of code for entropy loss calculation:

```python
entropy_loss = -th.mean(-log_prob)
```

However, this does not correctly represent the entropy of a probability distribution, which is a measure of the uncertainty or the randomness of a policy.

**Expected behavior:**
The entropy of a discrete probability distribution is defined in information theory as:

H(X) = -Σ p(x) * log(p(x))

where \( p(x) \) is the probability of the random variable \( X \) taking a specific value.

To encourage exploration in reinforcement learning, we want to maximize entropy. However, in optimization, we minimize a loss function. Therefore, we consider the negative entropy as the loss to be minimized.

**Actual behavior:**
The current code seems to calculate the negative of the mean of the negative log probabilities, which is not the same as entropy. It fails to weight the log probabilities by their respective probabilities, which is a critical part of the entropy calculation.

**Steps/code to reproduce the bug:**
Provide a snippet of code or describe the situation in which the entropy loss calculation does not work as expected.

**Suggested fix:**
The entropy loss should be calculated using the actual probabilities and their log probabilities as follows:

```python
entropy_loss = -th.mean(th.exp(log_prob) * log_prob)
```

Here, `th.exp(log_prob)` converts the log probabilities back to probabilities \( p(x) \), and `log_prob` is the log probability \( \log p(x) \). The product of these yields the terms of the entropy sum.

Alternatively, if the policy network provides the probabilities directly, the entropy can be calculated as:

```python
entropy_loss = -th.sum(prob * th.log(prob))
```

where `prob` is the probability of each action.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.
